app/models/declaration.rb

```diff
@@ -1,20 +1,142 @@
 class Declaration < ApplicationRecord
-  enum :declaration_type,
-       {
-         "started" => "started",
-         "retained-1" => "retained-1",
-         "retained-2" => "retained-2",
-         "retained-3" => "retained-3",
-         "retained-4" => "retained-4",
-         "completed" => "completed",
-         "extended-1" => "extended-1",
-         "extended-2" => "extended-2",
-         "extended-3" => "extended-3"
-       },
-       validate: { message: "Choose a valid declaration type" }
-
   belongs_to :training_period
+  belongs_to :voided_by_user, class_name: "User", optional: true
+  belongs_to :mentor_teacher, class_name: "Teacher", optional: true
   has_many :statement_line_items, class_name: "Statement::LineItem"
 
+  enum :status, {
+    submitted: "submitted",
+    eligible: "eligible",
+    payable: "payable",
+    paid: "paid",
+    voided: "voided",
+    ineligible: "ineligible",
+    awaiting_clawback: "awaiting_clawback",
+    clawed_back: "clawed_back"
+  }, validate: { message: "Choose a valid status" }, suffix: true
+
+  enum :declaration_type, {
+    started: "started",
+    "retained-1": "retained-1",
+    "retained-2": "retained-2",
+    "retained-3": "retained-3",
+    "retained-4": "retained-4",
+    "extended-1": "extended-1",
+    "extended-2": "extended-2",
+    "extended-3": "extended-3",
```

Comment:
Doesn't it end up being a bit awkward having symbol hash keys with dashes? i.e., `thing[:"retained-1"]`